src/recipe.rs

```diff
@@ -300,6 +300,15 @@ impl<'src, D> Recipe<'src, D> {
           }
         }
         Err(io_error) => {
+          if let Some(working_directory) = self.working_directory(context) {
+            if let Err(io_error) = fs::read_dir(working_directory.clone()) {
```

Comment:
This can probably pass a reference to avoid cloning:

```suggestion
            if let Err(io_error) = fs::read_dir(&working_directory) {
```